src/System.Text.Json/tests/Utf8JsonReaderTests.cs

```diff
@@ -1738,6 +1751,69 @@ public static void ConsumeLotsOfComments(string valueString, bool insideArray, s
             Assert.Equal(dataUtf8.Length, json.BytesConsumed);
         }
 
+        [Theory]
+        [MemberData(nameof(SingleLineCommentData))]
+        public static void ConsumeSingleLineCommentSingleSpanTest(string expected, string[] inputData)
+        {
+            var state = new JsonReaderState(options: new JsonReaderOptions { CommentHandling = JsonCommentHandling.Allow });
+            int bytesConsumed = 0;
+            var dataUtf8 = new byte[] { };
+
+            for (int count = 0; count < inputData.Length; ++count)
+            {
+                bool isFinal = (count == inputData.Length - 1);
+                dataUtf8 = dataUtf8.Concat(Encoding.UTF8.GetBytes(inputData[count])).ToArray();
+                var json = new Utf8JsonReader(dataUtf8, isFinalBlock: isFinal, state);
+                while (json.Read())
+                {
+                    switch (json.TokenType)
+                    {
+                        case JsonTokenType.StartObject:
+                        case JsonTokenType.EndObject:
+                            break;
+                        case JsonTokenType.Comment:
+                            Assert.Equal(Encoding.UTF8.GetBytes(expected), json.ValueSpan.ToArray());
+                            break;
+                        default:
+                            Assert.True(false);
+                            break;
+                    }
+                }
+                dataUtf8 = dataUtf8.Skip(bytesConsumed).ToArray();
```

Comment:
bytesConsumed is never set to anything other than 0. Should this be `json.BytesConsumed`?

Reply:
oops.